agent/containermetadata/parse_metadata.go

```diff
@@ -0,0 +1,153 @@
+// and packages this data for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseDockerContainerMetadata(task *api.Task, container *api.Container, dockerContainer *docker.Container) DockerContainerMetadata {
+	if container == nil {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: container metadata not available or does not exist", task, container)
+		return DockerContainerMetadata{}
+	}
+
+	// In most cases a container should never lack a config but we check regardless to avoid
+	// nil pointer exceptions (Could occur if there is some error in the docker api call, if the
+	// container we receive has incomplete information)
+	imageNameFromConfig := ""
+	if dockerContainer.Config != nil {
+		imageNameFromConfig = dockerContainer.Config.Image
+	} else {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: container has no configuration", task, container)
+	}
+
+	// Get Port bindings from docker configurations
+	var ports []api.PortBinding
+	var err error
+	if dockerContainer.HostConfig != nil {
+		ports, err = api.PortBindingFromDockerPortBinding(dockerContainer.HostConfig.PortBindings)
+		if err != nil {
+			seelog.Warnf("Failed to parse container metadata for task %s container %s: %v", task, container, err)
+		}
+	} else {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: container has no host configuration", task, container)
+	}
+
+	networkMetadata, err := parseNetworkMetadata(dockerContainer.NetworkSettings, dockerContainer.HostConfig)
+	if err != nil {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: %v", task, container, err)
+	}
+
+	return DockerContainerMetadata{
+		containerID:         dockerContainer.ID,
+		dockerContainerName: dockerContainer.Name,
+		imageID:             dockerContainer.Image,
+		imageName:           imageNameFromConfig,
+		ports:               ports,
+		networkInfo:         networkMetadata,
+	}
+}
+
+// parseTaskMetadata parses  in the AWS configuration and task
+// and packages this data for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseTaskMetadata(task *api.Task, container *api.Container) TaskMetadata {
+	containerName := container.Name
+
+	taskARNFromConfig := task.Arn
+
+	return TaskMetadata{
+		containerName: containerName,
+		taskARN:       taskARNFromConfig,
+	}
+}
+
+// parseMetadata gathers metadata from a docker container, and task
+// configuration and data then packages it for JSON Marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this or the above stage.
+func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, task *api.Task, container *api.Container) Metadata {
+	taskMD := parseTaskMetadata(task, container)
```

Comment:
Along the same lines, I don't see the value of creating a function which just initializes a struct. It would be lot helpful to anyone who reads the code to just do this instead:
` return Metadata { taskMetadata: TaskMetadata{...} ...}`